src/idt.rs

```diff
@@ -326,6 +326,6 @@ pub(crate) fn init() {
     unsafe {
         static mut IDT: Idt = Idt::empty();
         IDT.init();
-        lidt(&*core::ptr::addr_of!(IDT));
+        IDT.load();
```

Comment:
Hmm, I'm not actually sure if this change is loyal to the _spirit_ of the "static mut refs" lint. The [intention of that lint is that references (`&`s and `&mut`s) should never be created to `static mut`s](https://github.com/rust-lang/rust/issues/114447). Previously, the reference was created explicitly using the `&` operator. Now that `load` is changed to a method on `Idt` with a `&'static mut self` receiver, the reference is implicitly by the method call. However, a reference *is* still being created. If we're no longer getting warnings after this change, that means that the compiler apparently doesn't know how to warn about this case...but we're still doing the thing that the warning is intended to tell us not to do. 

So, this might be acceptable, if we think the deprecation of references to `static mut`s is dumb and we just want to make the warning go away. But, if our intention is to be loyal to the *spirit* of that warning, we probably want to change the signature of `Idt::load` from

```rust
unsafe fn load(&'static mut self)
```

to 

```rust
unsafe fn load(idt: *const Idt)
```

and change this code to:
```suggestion
        Idt::load(core::ptr::addr_of!(IDT));
```

This way, no reference is ever created to `IDT`, and we instead interact with it exclusively through raw pointers (provided that we make a similar change to the `Idt::init` function).

Reply:
Hmm, doesn't this imply that we can never call a method on a `static mut` unless it's by value? That seems overly restrictive.

Reply:
AFAICT from reading through the Rust issue, yes, the intention is that calling methods on a `static mut` is now frowned upon. See e.g. https://github.com/rust-lang/rust/issues/114447#issuecomment-1837178810

Reply:
That's a huge bummer.

My gut is to get this in now (or just leave it as is), and then work with the upstream community to see how this takes shape going forward.

Going to using raw pointers here is a huge step backwards, as they do nothing to convey any sense of ownership or validity. In _this_ case it's probably fine, since the code is compact enough that we can just look at it and just sorta "know" that the pointers are correct, but still.

Reply:
Btw, it's telling that a lot of the fixes that seem to be coming in addressing these warnings are just `&*core::ptr::addr_of!(STATICY_THING);`

Sigh.

Reply:
Hmm.... I wanted to look into using `OnceCell` for this, but that isn't `Sync`. Sigh.

Reply:
Ok, with the introduction of `SyncUnafeCell` for the IDT, I think this is good to go. Eliza, do you have any other concerns here?